### PR TITLE
PREAPPS-5640: Versioning files in Briefcase

### PR DESCRIPTION
**Pull request by miteshsavani810**

### CODE REVIEW

File: src/batch-client/index.ts [hunk 1]
```diff
@@ -1352,6 +1353,29 @@ export class ZimbraBatchClient {
...
+				doc: {
+					id,
+					ver: version,
+					count
+				}
+			}
+		}).then(response => {
+			const data = normalize(ListDocumentRevisions)(response);
+			const versionIds = data.documents.map((doc: any) => doc.version);
+			const maxVersion = Math.max(...versionIds);
```
Comment (silentsakky) [inline src/batch-client/index.ts:1370]: I think version ids are in string format, how does Math.max works with string data? does it convert to int implicitly?
for clarity we should convert these version ids to int ourself

File: src/batch-client/index.ts [hunk 2]
```diff
@@ -1352,6 +1353,29 @@ export class ZimbraBatchClient {
...
+			const latestDocumentVersion = data.documents.find((doc: any) => doc.version === maxVersion);
```
Comment (silentsakky) [inline src/batch-client/index.ts:1371]: is maxVersion a string or int?

File: src/batch-client/index.ts [hunk 3]
```diff
@@ -1352,6 +1353,29 @@ export class ZimbraBatchClient {
...
+
+			return {
+				...latestDocumentVersion,
+				subchilds: data.documents
```
Comment (silentsakky) [inline src/batch-client/index.ts:1375]: instead of subchilds call it docs

File: src/schema/generated-schema-types.ts [hunk 4]
```diff
@@ -3530,6 +3537,13 @@ export type QueryGetWorkingHoursArgs = {
 };
 
 
+export type QueryListDocumentRevisionsArgs = {
+  count: Scalars['String'];
```
Comment (silentsakky) [inline src/schema/generated-schema-types.ts:3541]: should be a number?

File: src/schema/generated-schema-types.ts [hunk 5]
```diff
@@ -3530,6 +3537,13 @@ export type QueryGetWorkingHoursArgs = {
...
+  id: Scalars['ID'];
+  version: Scalars['String'];
```
Comment (silentsakky) [inline src/schema/generated-schema-types.ts:3543]: should be a number?

File: src/batch-client/index.ts (hunk 1 above)
Comment (miteshsavani810) [reply src/batch-client/index.ts:1370]: Version ids are in Int format.

File: src/batch-client/index.ts (hunk 2 above)
Comment (miteshsavani810) [reply src/batch-client/index.ts:1371]: it's int.

File: src/schema/generated-schema-types.ts (hunk 4 above)
Comment (miteshsavani810) [reply src/schema/generated-schema-types.ts]: Done.

File: src/schema/generated-schema-types.ts (hunk 5 above)
Comment (miteshsavani810) [reply src/schema/generated-schema-types.ts]: Done.

**Review by silentsakky — APPROVED** [review]

**Pull request merged**